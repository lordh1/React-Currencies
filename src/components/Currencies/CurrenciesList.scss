.CurrenciesList {
  &__header {
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.2), 0px 2px 2px rgba(0, 0, 0, 0.12), 0px 0px 2px rgba(0, 0, 0, 0.14);
    border-radius: 3px;
    padding: 10px;
    height: 30px;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    text-align: center;
    justify-content: center;
    letter-spacing: 0.15px;
    text-transform: uppercase;

    & * {
      width: 25%;

      @media screen and (max-width: 500px) {
        flex: 0 0 100%;
      }
    }

    @media screen and (max-width: 500px) {
      height: 150px;
    }
  }
}
