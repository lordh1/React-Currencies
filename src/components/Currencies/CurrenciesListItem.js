import React from 'react';
import './CurrenciesListItem.scss';
import PropTypes from 'prop-types';

const CurrenciesListItem = ({ data, no }) => {
  return (
    <div className="CurrenciesListItem">
      <div className="CurrenciesListItem__column">{no}</div>
      <div className="CurrenciesListItem__column">{data.currency}</div>
      <div className="CurrenciesListItem__column">{data.code}</div>
      <div className="CurrenciesListItem__column">{data.mid}</div>
    </div>
  );
};

CurrenciesListItem.propTypes = {
  data: PropTypes.shape({
    currency: PropTypes.string.isRequired,
    code: PropTypes.string.isRequired,
    mid: PropTypes.number.isRequired,
  }).isRequired,
  no: PropTypes.number.isRequired,
};

export default CurrenciesListItem;
