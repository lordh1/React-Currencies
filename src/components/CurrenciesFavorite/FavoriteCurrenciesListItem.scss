.FavoriteCurrenciesListItem {
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.2), 0px 2px 2px rgba(0, 0, 0, 0.12), 0px 0px 2px rgba(0, 0, 0, 0.14);
    border-radius: 3px;
    height: auto;
    margin: 5px;
    align-items: center;
    vertical-align: top;
    resize: none;
    box-sizing: border-box;
    color: #a0a0a0;
    font-family: Roboto, serif;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 17px;
    letter-spacing: 0.15px;
    padding: 5px;
    display: inline-flex;

    &:hover {
        cursor: pointer;
        font-weight: bold;
    }

    &__active {
        background-color: #0eb6ff;
        color: #000000;
    }
}
