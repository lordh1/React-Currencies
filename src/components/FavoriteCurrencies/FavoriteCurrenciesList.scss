.FavoriteCurrenciesList {
  margin: 20px;
}
