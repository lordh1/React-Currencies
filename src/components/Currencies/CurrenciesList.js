import * as React from "react";
import PropTypes from "prop-types";
import CurrenciesListItem from "./CurrenciesListItem";
import "./CurrenciesList.scss";
import { useEffect } from "react";
import { useState } from "react";

const CurrenciesList = ({ currencies, favoriteCurrencies }) => {
  const [currenciesList, setCurrenciesList] = useState([]);

  useEffect(() => {
    const list = [...currencies.rates];
    const filteredList = list.filter((item) =>
      favoriteCurrencies.includes(item.code)
    );
    setCurrenciesList(filteredList);
  }, [favoriteCurrencies, currencies]);

  return (
    <div className="CurrenciesList">
      <h2>Rates of your favorite currencies</h2>
      {currenciesList.length ? (
        <div className="CurrenciesList__header">
          <div>No</div>
          <div>Currency</div>
          <div>Code</div>
          <div>Mid</div>
        </div>
      ) : (
        <h4>You need to follow any currency</h4>
      )}
      {currenciesList.map((item, key) => (
        <CurrenciesListItem
          data={item}
          key={`${key.toString()}${item.code}`}
          no={key + 1}
        />
      ))}
    </div>
  );
};

CurrenciesList.propTypes = {
  currencies: PropTypes.object,
  favoriteCurrencies: PropTypes.array
};

export default CurrenciesList;
