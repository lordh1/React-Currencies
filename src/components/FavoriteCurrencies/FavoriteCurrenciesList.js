import * as React from "react";
import FavoriteCurrenciesListItem from "./FavoriteCurrenciesListItem";
import "./FavoriteCurrenciesList.scss";
import PropTypes from "prop-types";

const FavoriteCurrenciesList = ({ currencies, favoriteCurrencies, onClick }) => {
  const checkIsActive = (code) => {
    return !!favoriteCurrencies.includes(code);
  };

  return (
    <div className="FavoriteCurrenciesList">
      <h2>Please click on the currency to follow/unfollow it</h2>
      {currencies.rates.map((item, key) => (
        <FavoriteCurrenciesListItem
          data={item}
          key={`${key.toString()}${item.code}`}
          no={key + 1}
          onClick={onClick}
          isActive={checkIsActive(item.code)}
        />
      ))}
    </div>
  );
};

FavoriteCurrenciesList.propTypes = {
  currencies: PropTypes.object,
  favoriteCurrencies: PropTypes.array,
  onClick: PropTypes.func
};

export default FavoriteCurrenciesList;
