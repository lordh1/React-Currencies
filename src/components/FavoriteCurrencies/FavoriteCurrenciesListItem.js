import React from "react";
import "./FavoriteCurrenciesListItem.scss";
import PropTypes from "prop-types";

const FavoriteCurrenciesListItem = ({ data, onClick, isActive }) => {
  return (
    <span
      onClick={() => onClick(data.code)}
      className={`${"FavoriteCurrenciesListItem"}
        ${isActive ? "FavoriteCurrenciesListItem__active" : ""}`}
    >
      {data.code}
    </span>
  );
};

FavoriteCurrenciesListItem.propTypes = {
  data: PropTypes.object,
  onClick: PropTypes.func,
  isActive: PropTypes.bool
};

export default FavoriteCurrenciesListItem;
