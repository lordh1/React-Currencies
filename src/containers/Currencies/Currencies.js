import * as React from "react";
import { useDispatch, useMappedState } from "redux-react-hook";
import "./Currencies.scss";
import CurrenciesList from "../../components/Currencies/CurrenciesList";
import FavoriteCurrenciesList from "../../components/CurrenciesFavorite/FavoriteCurrenciesList";
import * as favoriteCurrenciesTypes from "../../store/favoriteCurrencies/types";

const mapState = (state) => ({
  currencies: state.currencies,
  favoriteCurrencies: state.favoriteCurrencies
});

const Currencies = () => {
  const dispatch = useDispatch();

  const { currencies, favoriteCurrencies } = useMappedState(mapState);

  const favoriteCurrenciesToggle = (code) => {
    dispatch({
      type: favoriteCurrenciesTypes.FAVORITE_CURRENCIES_TOGGLE,
      code: code
    });
  };

  return (
    <div className="Currencies">
      <h1>Currencies</h1>
      <FavoriteCurrenciesList
        currencies={currencies}
        favoriteCurrencies={favoriteCurrencies}
        onClick={favoriteCurrenciesToggle}
      />
      <CurrenciesList
        currencies={currencies}
        favoriteCurrencies={favoriteCurrencies}
      />
    </div>
  );
};

export default Currencies;
