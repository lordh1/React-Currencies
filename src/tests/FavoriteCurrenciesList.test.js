import React from 'react';
import { mount } from 'enzyme';
import { StoreContext } from 'redux-react-hook';
import configureStore from 'redux-mock-store';
import FavoriteCurrenciesList from '../components/FavoriteCurrencies/FavoriteCurrenciesList';

import FavoriteCurrenciesListItem from '../components/FavoriteCurrencies/FavoriteCurrenciesListItem';
import Currencies from '../containers/Currencies/Currencies';

const currencies = {
  table: 'A',
  no: '007/A/NBP/2020',
  effectiveDate: '2020-01-13',
  rates: [
    { currency: 'bat (Tajlandia)', code: 'THB', mid: 0.1261 },
    { currency: 'dolar amerykański', code: 'USD', mid: 3.8101 },
    { currency: 'dolar australijski', code: 'AUD', mid: 2.6312 },
    { currency: 'dolar Hongkongu', code: 'HKD', mid: 0.4903 },
    { currency: 'dolar kanadyjski', code: 'CAD', mid: 2.9176 },
    { currency: 'dolar nowozelandzki', code: 'NZD', mid: 2.5274 },
    { currency: 'dolar singapurski', code: 'SGD', mid: 2.829 },
    { currency: 'euro', code: 'EUR', mid: 4.2353 },
    { currency: 'forint (Węgry)', code: 'HUF', mid: 0.012645 },
    { currency: 'frank szwajcarski', code: 'CHF', mid: 3.917 },
    { currency: 'funt szterling', code: 'GBP', mid: 4.9432 },
    { currency: 'hrywna (Ukraina)', code: 'UAH', mid: 0.1593 },
    { currency: 'jen (Japonia)', code: 'JPY', mid: 0.034687 },
    { currency: 'korona czeska', code: 'CZK', mid: 0.1678 },
    { currency: 'korona duńska', code: 'DKK', mid: 0.5667 },
    { currency: 'korona islandzka', code: 'ISK', mid: 0.03087 },
    { currency: 'korona norweska', code: 'NOK', mid: 0.4285 },
    { currency: 'korona szwedzka', code: 'SEK', mid: 0.4006 },
    { currency: 'kuna (Chorwacja)', code: 'HRK', mid: 0.5688 },
    { currency: 'lej rumuński', code: 'RON', mid: 0.886 },
    { currency: 'lew (Bułgaria)', code: 'BGN', mid: 2.1655 },
    { currency: 'lira turecka', code: 'TRY', mid: 0.6501 },
    { currency: 'nowy izraelski szekel', code: 'ILS', mid: 1.0985 },
    { currency: 'peso chilijskie', code: 'CLP', mid: 0.004925 },
    { currency: 'peso filipińskie', code: 'PHP', mid: 0.0755 },
    { currency: 'peso meksykańskie', code: 'MXN', mid: 0.2029 },
    {
      currency: 'rand (Republika Południowej Afryki)',
      code: 'ZAR',
      mid: 0.2653,
    },
    { currency: 'real (Brazylia)', code: 'BRL', mid: 0.9302 },
    { currency: 'ringgit (Malezja)', code: 'MYR', mid: 0.938 },
    { currency: 'rubel rosyjski', code: 'RUB', mid: 0.0625 },
    { currency: 'rupia indonezyjska', code: 'IDR', mid: 0.00027879 },
    { currency: 'rupia indyjska', code: 'INR', mid: 0.053836 },
    { currency: 'won południowokoreański', code: 'KRW', mid: 0.003295 },
    { currency: 'yuan renminbi (Chiny)', code: 'CNY', mid: 0.5531 },
    { currency: 'SDR (MFW)', code: 'XDR', mid: 5.2674 },
  ],
};
const favoriteCurrencies = ['USD', 'EUR'];

const mockStore = configureStore();

describe('Currencies', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore({
      currencies,
      favoriteCurrencies,
    });

    store.dispatch = jest.fn();

    wrapper = mount(
      <StoreContext.Provider value={store}>
        <Currencies />
      </StoreContext.Provider>,
    );
  });

  it('FavoriteCurrenciesList component exists', () => {
    const item = wrapper.find(FavoriteCurrenciesList);
    expect(item).toHaveLength(1);
  });

  it('FavoriteCurrenciesListItem components count is 35', () => {
    const item = wrapper.find(FavoriteCurrenciesListItem);
    expect(item).toHaveLength(35);
  });

  it('FavoriteCurrenciesListItem - THB currency exists', () => {
    const item = wrapper.find(FavoriteCurrenciesListItem);
    expect(item.at(0).text()).toEqual('THB');
  });

  it('FavoriteCurrenciesListItem - THB currency is active', () => {
    const item = wrapper.find(FavoriteCurrenciesListItem);
    expect(item.at(0).props().isActive).toEqual(false);
  });

  it('FavoriteCurrenciesListItem - EUR currency exists', () => {
    const item = wrapper.find(FavoriteCurrenciesListItem);
    expect(item.at(1).text()).toEqual('USD');
  });

  it('FavoriteCurrenciesListItem - EUR currency is active', () => {
    const item = wrapper.find(FavoriteCurrenciesListItem);
    expect(item.at(1).props().isActive).toEqual(true);
  });

  it('matches snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
