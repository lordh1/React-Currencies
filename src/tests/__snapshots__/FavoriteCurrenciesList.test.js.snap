// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library matches snapshot 1`] = `
"<Currencies>
  <div className=\\"Currencies\\">
    <h1>
      Currencies
    </h1>
    <FavoriteCurrenciesList currencies={{...}} favoriteCurrencies={{...}} onClick={[Function: favoriteCurrenciesToggle]}>
      <div className=\\"FavoriteCurrenciesList\\">
        <h2>
          Please click on the currency to follow/unfollow it
        </h2>
        <FavoriteCurrenciesListItem data={{...}} no={1} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            THB
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={2} onClick={[Function: favoriteCurrenciesToggle]} isActive={true}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      FavoriteCurrenciesListItem__active\\" role=\\"button\\" tabIndex=\\"0\\">
            USD
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={3} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            AUD
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={4} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            HKD
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={5} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            CAD
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={6} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            NZD
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={7} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            SGD
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={8} onClick={[Function: favoriteCurrenciesToggle]} isActive={true}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      FavoriteCurrenciesListItem__active\\" role=\\"button\\" tabIndex=\\"0\\">
            EUR
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={9} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            HUF
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={10} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            CHF
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={11} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            GBP
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={12} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            UAH
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={13} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            JPY
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={14} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            CZK
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={15} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            DKK
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={16} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            ISK
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={17} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            NOK
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={18} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            SEK
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={19} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            HRK
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={20} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            RON
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={21} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            BGN
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={22} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            TRY
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={23} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            ILS
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={24} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            CLP
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={25} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            PHP
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={26} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            MXN
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={27} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            ZAR
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={28} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            BRL
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={29} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            MYR
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={30} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            RUB
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={31} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            IDR
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={32} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            INR
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={33} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            KRW
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={34} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            CNY
          </span>
        </FavoriteCurrenciesListItem>
        <FavoriteCurrenciesListItem data={{...}} no={35} onClick={[Function: favoriteCurrenciesToggle]} isActive={false}>
          <span onClick={[Function: onClick]} onKeyPress={[Function: onKeyPress]} className=\\"FavoriteCurrenciesListItem\\\\n      \\" role=\\"button\\" tabIndex=\\"0\\">
            XDR
          </span>
        </FavoriteCurrenciesListItem>
      </div>
    </FavoriteCurrenciesList>
    <CurrenciesList currencies={{...}} favoriteCurrencies={{...}}>
      <div className=\\"CurrenciesList\\">
        <h2>
          Rates of your favorite currencies
        </h2>
        <div className=\\"CurrenciesList__header\\">
          <div>
            No
          </div>
          <div>
            Currency
          </div>
          <div>
            Code
          </div>
          <div>
            Mid
          </div>
        </div>
        <CurrenciesListItem data={{...}} no={1}>
          <div className=\\"CurrenciesListItem\\">
            <div className=\\"CurrenciesListItem__column\\">
              1
            </div>
            <div className=\\"CurrenciesListItem__column\\">
              dolar amerykański
            </div>
            <div className=\\"CurrenciesListItem__column\\">
              USD
            </div>
            <div className=\\"CurrenciesListItem__column\\">
              3.8101
            </div>
          </div>
        </CurrenciesListItem>
        <CurrenciesListItem data={{...}} no={2}>
          <div className=\\"CurrenciesListItem\\">
            <div className=\\"CurrenciesListItem__column\\">
              2
            </div>
            <div className=\\"CurrenciesListItem__column\\">
              euro
            </div>
            <div className=\\"CurrenciesListItem__column\\">
              EUR
            </div>
            <div className=\\"CurrenciesListItem__column\\">
              4.2353
            </div>
          </div>
        </CurrenciesListItem>
      </div>
    </CurrenciesList>
  </div>
</Currencies>"
`;
